orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -592,6 +604,42 @@ def run_step(model1, model2, model3, x1, x2):
         _test_helpers.assert_gradients_match_and_reset_gradient(ort_model2, pt_model2)
         _test_helpers.assert_gradients_match_and_reset_gradient(ort_model3, pt_model3)
 
+def test_ortmodule_inputs_with_dynamic_shape():
+    D_in, H, D_out = 784, 500, 10
+
+    model = NeuralNetSinglePositionalArgument(D_in, H, D_out).to('cuda')
+    model = ORTModule(model)
+
+    for step in range(10):
+        N = random.randint(1,100)
+        x = torch.randn(N, D_in, device='cuda', requires_grad=True)
+        assert x.grad is None
+
+        prediction = model(x)
+        s = prediction.sum()
+        s.backward()
+
+        assert x.grad is not None 
+        for param in model.parameters():
+            assert param.grad is not None
+            param.grad = None
```

Comment:
please add asserts for forward results and gradient correctness. 
see other "correctness" UTs for example.

Reply:
assert gradient is not None is too week :)